File: examples/python/switch-blink.py
import RPi.GPIO as GPIO
import time

pinIn = 18
pinOut = 17

GPIO.setmode(GPIO.BCM)
GPIO.setwarnings(False)
GPIO.setup(pinIn, GPIO.IN, pull_up_down = GPIO.PUD_DOWN)
GPIO.setup(pinOut, GPIO.OUT)

def blink(channel):
	print("Button pressed")
	GPIO.output(pinOut,GPIO.HIGH)
	time.sleep(0.2)
	GPIO.output(pinOut,GPIO.LOW)

GPIO.add_event_detect(pinIn, GPIO.RISING, callback=blink, bouncetime=300)

while True:
	time.sleep(1)

GPIO.cleanup()

File: ruby/switch-blink.rb
require 'pi_piper'
include PiPiper

pin_in_num = 18
pin_out = PiPiper::Pin.new(:pin => 17, :direction => :out)

def blink(pin)
	puts "Button pressed"
	pin.on
	sleep(0.2)
	pin.off
end

after :pin => pin_in_num, :goes => :high do
	blink(pin_out)
end

PiPiper.wait